Client/Frontend/Toolbar+URLBar/StatefulButton.swift

```diff
@@ -0,0 +1,45 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+
+enum ReloadButtonState: String {
+    case reload = "Reload"
+    case stop = "Stop"
+    case disabled = "Disabled"
+}
+
+class StatefulButton: UIButton {
+    convenience init(frame: CGRect, state: ReloadButtonState) {
+        self.init(frame: frame)
+        reloadButtonState = state
+    }
+
+    required override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    var _reloadButtonState = ReloadButtonState.disabled
```

Comment:
```suggestion
    // MARK: - Variables
    
    var _reloadButtonState = ReloadButtonState.disabled
```

Also, same comment here re the `_reload...` syntax for data encapsulation